src/Shared/SearchWrapper.js

```diff
@@ -0,0 +1,123 @@
+import React, { Component } from 'react'
+import {
+  View,
+  Text,
+  StyleSheet,
+  Image,
+  TextInput,
+  Platform,
+} from 'react-native'
+import Icon from 'react-native-vector-icons/dist/MaterialIcons'
+
+export default class SearchBarWrapper extends Component {
+  debounce = (fn, time) => {
+    let timeout
+
+    return function () {
+      const functionCall = () => fn.apply(this, arguments)
+
+      clearTimeout(timeout)
+      timeout = setTimeout(functionCall, time)
+    }
+  }
+
+  render() {
+    let { searchBar, onFilterElement, notFound, notFoundText, children } =
+      this.props
+
+    if (!searchBar) {
+      return <>{children}</>
+    }
+
+    if (searchBar.enabled) {
+      return (
+        <View style={styles.totalWrapper}>
+          <View
+            style={[
+              styles.searchBar,
+              {
+                backgroundColor: searchBar.backgroundColor,
```

Comment:
Generally, it's better to just destructure out all of those props from `searchBar`. That way you don't have repeat as much code.